### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by anxolerd**
I've restructured project a little.

- added `lib` module to store all helpers, utils, plugins and tools
- moved existing plugins to `lib.plugins` modulr
- decomposed utils and moved them to `lib.utils` module

Review request: @webknjaz 

Please, review **very carefully** because there is a merge in this PR, and something could go terribly wrong.

**PLEASE, DO NOT MAKE OTHER COMMITS TO `api-for-admin` BRANCH UNTIL THIS PR IS RESOLVED**

### CODE REVIEW

File: src/GDGUkraine/lib/utils/mail.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+import html2text
+from blueberrypy.template_engine import get_template
+
+logger = logging.getLogger(__name__)
+
+
+def gmail_send(oauth2session, message, sbj, to_email,
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: I've dropped `oauth2session` param. See 9f9fb7447c16a6f346026acf4534a0b95df88818.
Plz port this commit.

File: src/GDGUkraine/lib/utils/mail.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+               from_email='GDG Team Robot <kyiv@gdg.org.ua>'):
+
+    message['to'] = to_email
+    message['from'] = from_email
+    message['subject'] = sbj
+
+    st = oauth2session.post(
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: Same

File: src/GDGUkraine/lib/utils/mail.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                                                         .encode('utf8'))
+                        .decode('utf8')}),
+        headers={"content-type": "application/json"})
+
+    logger.debug(st.json())
+    logger.debug('Sent message to {}'.format(to_email))
+    return st.json()
+
+
+def gmail_send_html(oauth2session, template, payload, **kwargs):
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: Same

File: src/GDGUkraine/lib/utils/mail.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    msg = MIMEMultipart('alternative')
+
+    html_payload = get_template(template).render(**payload)
+
+    plain_text_payload = html2text.html2text(html_payload)
+
+    msg.attach(MIMEText(plain_text_payload, 'plain'))
+    msg.attach(MIMEText(html_payload, 'html'))
+
+    return gmail_send(oauth2session, message=msg, **kwargs)
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: Same

File: src/GDGUkraine/lib/utils/mail.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def gmail_send_text(oauth2session, payload, **kwargs):
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: Same

File: src/GDGUkraine/lib/utils/mail.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    msg = MIMEText(payload)
+
+    return gmail_send(oauth2session, message=msg, **kwargs)
```
Comment (webknjaz) [inline src/GDGUkraine/lib/utils/mail.py]: Same

File: src/GDGUkraine/rest_controller.py [hunk 7]
```diff
@@ -10,11 +10,13 @@
 from cherrypy.lib import httputil as cphttputil, file_generator
 from blueberrypy.util import from_collection, to_collection
 
-from .table_exporter import TableExporter
-
+from .lib.utils.table_exporter import TableExporter
```
Comment (webknjaz) [inline src/GDGUkraine/rest_controller.py]: I'd group similar exports together

File: src/GDGUkraine/rest_controller.py [hunk 8]
```diff
@@ -10,11 +10,13 @@
...
 from . import api
 from .model import User, Event, EventParticipant
-from .utils import gmail_send_html, aes_encrypt, make_vcard, pub
+
+from .lib.utils.mail import gmail_send_html
+from .lib.utils.vcard import make_vcard, aes_encrypt
+from .lib.utils.signals import pub
```
Comment (webknjaz) [inline src/GDGUkraine/rest_controller.py]: Same

**Comment by webknjaz** [comment]
@anxolerd Check my comments and also adjust newest changes in `rest_controller.py`

**Comment by anxolerd** [comment]
drop oauth2session bd7f991
group similar imports together 5f77be6

**Comment by anxolerd** [comment]
merge api-for-admin to make automatic merge possible e7f76d3

**Comment by webknjaz** [comment]
LGTM

**Pull request merged**